python/cuda_cccl/setup.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+#
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import os
+import shutil
+
+from setuptools import setup, find_namespace_packages
+
+
+project_path = os.path.abspath(os.path.dirname(__file__))
+cccl_path = os.path.abspath(os.path.join(project_path, "..", ".."))
+cccl_headers = [["cub", "cub"], ["libcudacxx", "include"], ["thrust", "thrust"]]
+__version__ = None
+with open(os.path.join(project_path, "cuda", "cccl", "_version.py")) as f:
+    exec(f.read())
+assert __version__ is not None
+ver = __version__
+del __version__
+
+
+with open("README.md") as f:
+    long_description = f.read()
+
+
+def copy_cccl_headers_to_cuda_include():
+    inc_path = os.path.join(project_path, "cuda", "_include")
```

Comment:
Important: Can we please establish the right layout before merging? There is a new wheel layout being communicated internally. This should be something like
```suggestion
    inc_path = os.path.join(project_path, "cccl", "include")
```

Reply:
Done: commit e3c78671258f9c41f2b0614e2566d20904ef94bb

I'm not sure about two aspects:

* Did you mean to suggest `site-packages/cccl/include/` as the  install directory? — I decided to make this `site-packages/cuda/cccl/include/`. It seemed odd to me to have it outside the cuda subdir.

* By accident I discovered that all cuda.cooperative unit tests pass without the CCCL headers. @gevtushenko for comment.

Reply:
> By accident I discovered that all cuda.cooperative unit tests pass without the CCCL headers. @gevtushenko for comment.

@rwgk we add path to CUDA Toolkit (CTK) headers here:
https://github.com/NVIDIA/cccl/blob/dcd8954e18ad5eaf61adef6241928f24db33187b/python/cuda_cooperative/cuda/cooperative/experimental/_nvrtc.py#L57

CTK provides CUB headers as well. Tests likely pass because they use CTK version of CUB. We should add a static assert somewhere in cuda.cooperative to check that we use version of CUB that cuda.cooperative was "build" with.

Reply:
Thanks @gevtushenko for that reminder! I solved this by factoring out `cuda.cccl.include_paths`. There is now this fragment (in include_paths.py):

```
    with as_file(files("cuda.cccl.include")) as f:
        cccl_incl = Path(f)
    assert cccl_incl.exists()
```

I believe it is very unlikely that this could still go wrong.

Reply:
`site-packages/cuda/cccl/include/{thrust,cub,cuda}` sounds right to me!